### PR TITLE
Use our actual feerate in open_channel messages, not a new one

### PR DESCRIPTION
**Pull request by valentinewallace**
When we were sending an open_channel messages we were asking the
feerate estimator for a new value instead of using the one we had.
If the feerate estimator gave a different value than the one it did
when we created the Channel struct, we'd start out-of-sync with our
counterparty and blow up on funding_signed. Even worse, the
ConfirmationTarget used was different, so its highly likely they
would disagree.

Also remove newly unused fee estimator parameter from get_open-channel
API.

### CODE REVIEW

**Comment by valentinewallace** [comment]
> Probably makes sense to swap our entire feerate_per_kw type globally for a u32. There is no reason to let the user-visible API return a type that we cannot use.
> 
> Also, needs a trivial full_stack_target fuzz fix since a previously-used fee-get is now unused.

Fixed!

File: lightning/src/ln/channel.rs [hunk 1]
```diff
@@ -1986,10 +1986,10 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		//If channel fee was updated by funder confirm funder can afford the new fee rate when applied to the current local commitment transaction
 		if update_fee {
 			let num_htlcs = local_commitment_tx.1;
-			let total_fee: u64 = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (num_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
+			let total_fee = feerate_per_kw * (COMMITMENT_TX_BASE_WEIGHT + (num_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) as u32 / 1000;
```
Comment (TheBlueMatt) [inline lightning/src/ln/channel.rs:1989]: Same here - probably cast up, not down.

File: lightning/src/ln/channel.rs [hunk 2]
```diff
@@ -3045,7 +3045,7 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			}
 		}
 
-		let proposed_sat_per_kw = msg.fee_satoshis * 1000 / closing_tx.get_weight() as u64;
+		let proposed_sat_per_kw = msg.fee_satoshis as u32 * 1000 / closing_tx.get_weight() as u32;
```
Comment (TheBlueMatt) [inline lightning/src/ln/channel.rs:3048]: I think this case is technically fine, but I'd feel better if we casted down after doing the arithmetic, not before.

File: lightning/src/ln/onchaintx.rs [hunk 3]
```diff
@@ -152,31 +152,31 @@ macro_rules! subtract_high_prio_fee {
 	($logger: ident, $fee_estimator: expr, $value: expr, $predicted_weight: expr, $used_feerate: expr) => {
 		{
 			$used_feerate = $fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::HighPriority);
-			let mut fee = $used_feerate * ($predicted_weight as u64) / 1000;
-			if $value <= fee {
+			let mut fee = $used_feerate * ($predicted_weight as u32) / 1000;
```
Comment (TheBlueMatt) [inline lightning/src/ln/onchaintx.rs:155]: Probably fine, but probably use a u64 (and would probably reduce diff size further down in the file).

**Pull request merged**